### ISSUE TITLE
CheckBox submits the incorrect value when checked and disabled

### ISSUE THREAD

**Issue by benrbrook** [danott/react-rails-form-helpers#31]
Hi! I noticed what I think is a bug in the `CheckBox` component. If you render a `CheckBox` that is both disabled and checked then submit the form, the checkbox's value gets passed back to the server as `0`, or unchecked.

For example:
```jsx
<CheckBox name='foo' checked={false} disabled />
```

produces something like

```html
<input type="hidden" readonly="" name="form_name[foo]" value="0"> // submitted
<input type="checkbox" name="form_name[foo]" disabled="" role="checkbox" id="form_name_foo" value="1" checked=""> // not submitted
```

The checkbox input is disabled, so it does not get submitted as part of the form. However, the hidden input is _not_ disabled, so it becomes the only thing submitted with a value that would correspond to being unchecked. It seems like the hidden input needs the `disabled` attribute at the very least. I can work around it by dynamically setting the `checkedValue` and `uncheckedValue` props based on `checked` but this feels like a hack.

**Comment by danott** [comment]
Yep, that is definitely a bug. Thank you for articulating it so clearly!

On March 21, 2022, GitHub ***@***.***> wrote:
> Hi! I noticed what I think is a bug in the CheckBox component. If you
> render a CheckBox that is both disabled and checked then submit the
> form, the checkbox's value gets passed back to the server as 0, or
> unchecked.
>
> For example:
>
> <CheckBox name='foo' checked={false} disabled />
> produces something like
>
> <input type="hidden" readonly="" name="form_name[foo]" value="0"> //
> submitted <input type="checkbox" name="form_name[foo]" disabled=""
> role="checkbox" id="form_name_foo" value="1" checked=""> // not
> submitted
> The checkbox input is disabled, so it does not get submitted as part
> of the form. However, the hidden input is not disabled, so it becomes
> the only thing submitted with a value that would correspond to being
> unchecked. It seems like the hidden input needs the disabled attribute
> at the very least. I can work around it by dynamically setting the
> checkedValue and uncheckedValue props based on checked but this feels
> like a hack.
>

—
Reply to this email directly, view it on GitHub
<https://github.com/danott/react-rails-form-helpers/issues/31>, or
unsubscribe <https://github.com/notifications/unsubscribe-
auth/AABNJBBUJVU637N4FCL5JV3VBC6LJANCNFSM5RISCOTA>.
You are receiving this because you are subscribed to this thread.Message
ID: ***@***.***>
